core/src/components/searchbar/searchbar.tsx

```diff
@@ -620,12 +629,19 @@ export class Searchbar implements ComponentInterface {
    * Clear button should be shown if one of two conditions applies:
    * 1. `showClearButton` is set to `always`.
    * 2. `showClearButton` is set to `focus`, and the searchbar has been focused.
+   * 3. `theme` is not set `ionic`, or the searchbar has not been disabled.
```

Comment:
```suggestion
   * Unless the `theme` is `ionic` and the searchbar is disabled.
```
Thinking more about this, this suggestion sounds easier to understand.